c/driver/common/utils.c

```diff
@@ -166,17 +166,23 @@ void AppendErrorDetail(struct AdbcError* error, const char* key, const uint8_t*
       return;
     }
 
-    memcpy(new_keys, details->keys, sizeof(char*) * details->count);
-    free(details->keys);
-    details->keys = new_keys;
+    if (details->keys != NULL) {
+      memcpy(new_keys, details->keys, sizeof(char*) * details->count);
```

Comment:
TIL that C does have a way of declaring a pointer parameter that is not NULL. I couldn't reproduce locally but these were failing on CI

Guessing the headers on the CI systems look something like:

https://stackoverflow.com/a/48830228/621736